src/generic/Map.jl

```diff
@@ -23,30 +23,21 @@ function preimage(f::Generic.CompositeMap{D, C}, a) where {D, C}
    return preimage(f.map1, preimage(f.map2, a))
 end
 
-function show(io::IO, ::MIME"text/plain", M::CompositeMap)
-   io = pretty(io)
-   println(io, "Composite map")
-   println(io, Indent(), "first ", Lowercase(), map1(M))
-   h = map2(M)
-   while h isa FunctionalCompositeMap || h isa CompositeMap
-     println(io, "next ", Lowercase(), h)
-     h = map2(h)
-   end
-   print(io, "next ", Lowercase(), h, Dedent())
+function AbstractAlgebra.show_map_data(io::IO, M::Union{CompositeMap, FunctionalCompositeMap})
+  println(io)
+  println(io, "which is the composite of", Indent())
+  println(io, map1(M))
+  print(io, map2(M))
 end
 
 function show(io::IO, M::CompositeMap)
    if get(io, :supercompact, false)
       # no nested printing
       print(io, "Composite map")
    else
-     io = pretty(io)
+      io = pretty(io)
       print(io, "Map: ", Lowercase(), domain(M))
-      h = map2(M)
-      while h isa FunctionalCompositeMap || h isa CompositeMap
```

Comment:
Note: I removed these loops, because in practice they did not work well, and e.g. `(f*g)*h` and `f*(g*h)` then printed differently (one showed two maps, one showed three maps). If we want, we can adjust the printing to show all "layers", recursively in a future PR -- but for now I think this is not very important. But I also didn't want to "slip this through" without notice